ci-automation/vendor-testing/aws.sh

```diff
@@ -0,0 +1,89 @@
+
+board="${CIA_ARCH}-usr"
+image_name="ci-${CIA_VERNUM}"
+aws_instance_type_var="AWS_${CIA_ARCH}_INSTANCE_TYPE"
+aws_instance_type="${!instance_type_var}"
+more_aws_instance_types_var="AWS_${CIA_ARCH}_MORE_INSTANCE_TYPES"
+more_aws_instance_types=( ${!more_instance_types_var} )
+
+if [[ "${AWS_AMI_ID}" == "" ]]; then
+    echo "++++ ${CIA_TESTSCRIPT}: downloading flatcar_production_ami_vmdk_image.vmdk.bz2 for ${CIA_VERNUM} (${CIA_ARCH}) ++++"
+    copy_from_buildcache "images/${CIA_ARCH}/${CIA_VERNUM}/flatcar_production_ami_vmdk_image.vmdk.bz2" .
+
+    lbunzip2 'flatcar_production_ami_vmdk_image.vmdk.bz2'
+
+    AWS_BUCKET="flatcar-kola-ami-import-${AWS_REGION}"
+    trap 'ore -d aws delete --region="${AWS_REGION}" --name="${image_name}" --ami-name="${image_name}" --file="flatcar_production_ami_vmdk_image.vmdk" --bucket "s3://${AWS_BUCKET}/${board}/"' EXIT
+    ore aws initialize --region="${AWS_REGION}" --bucket "${AWS_BUCKET}"
+    AWS_AMI_ID=$(ore aws upload --force --region="${AWS_REGION}" --name="${image_name}" --ami-name="${image_name}" --ami-description="Flatcar Test ${image_name}" --file='flatcar_production_ami_vmdk_image.vmdk' --bucket  "s3://${AWS_BUCKET}/${board}/" | jq -r .HVM)
+    echo "++++ ${CIA_TESTSCRIPT}: Created new AMI ${AWS_AMI_ID} (will be removed after testing) ++++"
+fi
+
+run_aws_kola_test() {
+    local instance_type="${1}"; shift
+    local instance_tapfile="${1}"; shift
+
+    timeout --signal=SIGQUIT 6h \
+        kola run \
+         --board="${board}" \
+         --basename="${image_name}" \
+         --channel="${CIA_CHANNEL}" \
+         --offering='basic' \
+         --parallel="${AWS_PARALLEL}" \
+         --platform=aws \
+         --aws-ami="${AWS_AMI_ID}" \
+         --aws-region="${AWS_REGION}" \
+         --aws-type="${instance_type}" \
+         --aws-iam-profile="${AWS_IAM_PROFILE}" \
+         --tapfile="${instance_tapfile}" \
+         --torcx-manifest=../torcx_manifest.json \
+         "${@}"
+}
+
+# these are set in ci-config.env
+export AWS_ACCESS_KEY_ID
+export AWS_SECRET_ACCESS_KEY
+
+extra_tests_run=0
+for instance_type in "${more_aws_instance_types[@]}"; do
+    instance_tests=( $(filter_prefixed_tests "${instance_type}" "${@}") )
+    if [[ "${#instance_tests}" -eq 1 ]] && [[ "${instance_tests[0]}" = '*' ]]; then
+        cl_internet_included="$(kola list --platform=aws --filter '*' | { grep cl.internet || : ; } )"
+        if [[ -n "${cl_internet_included}" ]]; then
+            instance_tests=( 'cl.internet' )
+        fi
+    fi
+    if [[ "${#instance_tests}" -gt 0 ]]; then
+        extra_tests_run=1
+        (
+            output=$(run_aws_kola_test "${instance_type}" "aws_validate_${instance_type}.tap" "${instance_tests[@]}" 2>&1 || :)
+            printf "=== START ${instance_type} ===\n%s\n=== END ${instance_type} ===\n" "$(sed -e 's/^/${instance_type}: /g' <<<"${output}")"
+            prefix_tap_file "${instance_type}" "aws_validate_${instance_type}.tap"
+        ) &
+    fi
+done
```

Comment:
https://github.com/flatcar-linux/scripts/pull/267 should have proper logic now for the cl.internet tests per instance. Should we merge it first and then factor it out into the sourced helper file in https://github.com/flatcar-linux/scripts/pull/293/files and then use it here?

Reply:
Overall I like the approach with the single for loop taken here, seems good but missing is for example that the final `run_kola_test` call only should be done if `main_tests` isn't empty.

Reply:
Yeah, thanks for spotting. I wrote all this filtering functionality with retesting in mind.